llvm/test/Transforms/LoopVectorize/AArch64/sve-runtime-check-size-based-threshold.ll

```diff
@@ -5,7 +5,6 @@ target triple = "aarch64-unknown-linux-gnu"
 
 ; Test case where the minimum profitable trip count due to runtime checks
 ; exceeds VF.getKnownMinValue() * UF.
-; FIXME: The code currently incorrectly is missing a umax(VF * UF, 28).
```

Comment:
I think this FIXME is wrong. We already generate the umax considering UF * VF. This is for the minimum iteration check, and UF*VF is needed for correctness.

the constant is the minimum iteration count needed to be profitable, but not for correctness. 

AFAICT the only reason the constant gets changed is due to the runtime checks being more expensive now, due to duplicated RT VF computations?

the actual memory checks seem to use the same runtime VF as originally?

Reply:
> AFAICT the only reason the constant gets changed is due to the runtime checks being more expensive now, due to duplicated RT VF computations?

If my understanding is correct, this constant is generated from the SCEVExpander call in addDiffRuntimeChecks.

>  This is for the minimum iteration check, and UF*VF is needed for correctness.

Yes, it's the profitability check that seems to be broken.